packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -517,6 +524,9 @@ export class ConnectionManager implements IConnectionManager {
             );
         }
 
+        if ((progress?.cancel?.aborted) ?? false) {
+            return;
```

Comment:
This comment likely applies to all other newly added early exit code paths as well (on some capacity).

I think you will be breaking some invariants here. For example, this.pendingConnection is not reset
Plus even more importantly - the connection should be closed. Not doing it means you are leaking a socket that may not close for hours!

I'd rather see this.setupNewSuccessfulConnection() being called, but it exists sooner - where we check for (this.closed) somehow.